tests/tests/test_configuration_deployments.py

```diff
@@ -132,3 +134,296 @@ def test_fail_duplicate_deployment(self, api_client_int, clean_db):
+            configuration_deployment,
+        )
+
+        dc = SimpleDeviceClient()
+        nextdep = dc.get_next_deployment(
+            dev.fake_token_mt(tenant_id),
+            artifact_name="dontcare",
+            device_type="hammer",
+        )
+        uri = nextdep.artifact["source"]["uri"]
+        qs = parse_qs(urlparse(uri).query)
+
+        # now break the url in various ways
+
+        # wrong deployment (signature error)
+        uri_bad_depl = uri.replace(deployment_id, str(uuid4()))
+        r = requests.get(
+            uri_bad_depl,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 403
+
+        # wrong tenant in url (signature error)
+        other_tenant_id = str(ObjectId())
+        _, r = api_client_int.create_tenant(other_tenant_id)
+        assert r.status_code == 201
+
+        uri_bad_tenant = uri.replace(tenant_id, other_tenant_id)
+        r = requests.get(
+            uri_bad_tenant,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 403
+
+        # wrong dev type (signature error)
+        other_dev = Device()
+        other_dev.device_type = "foo"
+        uri_bad_devtype = uri.replace(dev.device_type, other_dev.device_type)
+        r = requests.get(
+            uri_bad_devtype,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 403
+
+        # wrong dev id (signature error)
+        uri_bad_devid = uri.replace(dev.devid, other_dev.devid)
+        r = requests.get(
+            uri_bad_devid,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 403
+
+        # wrong x-men-signature
+        uri_bad_sig = uri.replace(
+            qs["x-men-signature"][0], "mftJRzBafnvMXhmMBH3THQertiEk0dZKP075bjBKccc"
+        )
+        r = requests.get(
+            uri_bad_sig,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 403
+
+        # no x-men-signature
+        uri_no_sig = uri.replace("&x-men-signature=", "")
+        r = requests.get(
+            uri_no_sig,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 400
+
+        # no x-men-expire
+        uri_no_exp = uri.replace("&x-men-expire=", "")
+        r = requests.get(
+            uri_no_exp,
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
+            verify=False,
+        )
+        assert r.status_code == 400
+
+        # GOTCHA: uri for one device can be accessed by a diff device from same tenant (jwt)
```

Comment:
self-comment:
@alfrunes these are some scenarios that surprised me.
it's all around authentication of the download endpoint.
let me know what you think!

Reply:
We can't use the authorization middleware for this endpoint, thus we have to pass all the parameters in the signed URL.
Note that this is also the case for S3 signed URLs.

Reply:
This reminds me I forgot to add the exception to the Træfik gateway: https://github.com/mendersoftware/integration/pull/1198

Reply:
ok, I guess it's clear now from the other comments - auth is not here by design, hence all questions are moot.